packages/cli/src/commands/execHandler.js

```diff
@@ -99,11 +99,30 @@ export const handler = async (args) => {
     ],
   })
 
-  try {
-    require.resolve(scriptPath)
-  } catch {
+  // check the script exists, and that there are no ambiguous conflicts like [foo.js, foo.ts]
+  const targetScriptPath = path.parse(scriptPath)
+  const targetScriptName = targetScriptPath?.base
+  const targetMatch = findScripts()
+    .map((p) => path.parse(p))
+    .filter((p) => {
+      if (!p || !targetScriptName) {
+        return false
+      }
+      // Compares with base when file specified as 'foo.ts', otherwise compares with name
+      // when file specified as 'foo'
+      return (targetScriptPath?.ext ? p.base : p.name) === targetScriptName
+    })
+  if (targetMatch.length !== 1) {
     console.error(
-      c.error(`\nNo script called ${c.underline(name)} in ./scripts folder.\n`)
+      c.error(
+        `\n${
+          !targetMatch.length ? 'No script' : 'Multiple scripts'
+        } called ${c.underline(
+          targetScriptPath?.ext
+            ? targetScriptPath.base
```

Comment:
You do `targetScriptPath?.` (with `?`) on the line above and below, but not here. Can `targetScriptPath` be undefined? If it can, we need `?` everywhere. If it can't, we can get rid of `?` everywhere 🙂

Reply:
Good question. I think in that particular instance `targetScriptPath` could not be undefined since the `targetScriptPath?.ext` condition already checked that it is defined, but that's not necessarily the case in the negative.

However, after re-reading the docs, I don't think it is possible for [path.parse()](https://nodejs.org/api/path.html#pathparsepath) to return undefined, so I'll remove all the optional chaining from those types.